src/lib/core/overlay/overlay-ref.ts

```diff
@@ -101,18 +103,28 @@ export class OverlayRef implements PortalHost {
     let backdropToDetach = this._backdropElement;
 
     if (backdropToDetach) {
-      backdropToDetach.classList.remove('md-overlay-backdrop-showing');
-      backdropToDetach.classList.remove(this._state.backdropClass);
-      backdropToDetach.addEventListener('transitionend', () => {
-        backdropToDetach.parentNode.removeChild(backdropToDetach);
+      let onTransitionEnd = () => {
```

Comment:
Since this can be used in two ways now, we should rename it to something like `finishDetach`